### DISCUSSION TITLE
larger than 20mb hard drives

### DISCUSSION THREAD

**Discussion by darvin** [dbalsom/marty#22] [Ideas]
- creation of vhd image from menu
 - support in the hard disk controller

what would be the good target for implemention? perhaps the controller the is already implemented - IBM/Xebec

which [variation](https://minuszerodegrees.net/ibm_xebec/ibm_xebec.htm) of the Xebec adaptor was used? if [this](https://www.ebay.com/itm/325444473347) one, was did it support any hard drives over 20mb? or its better to implement [this](https://www.ebay.com/itm/325292311821) one?

## [from](https://minuszerodegrees.net/ibm_xebec/ibm_xebec.htm)

Click [here](https://minuszerodegrees.net/oa/OA%20-%20IBM%20Fixed%20Disk%20Adapter.pdf) to see IBM's technical document for this variation of the card (contains matching circuit diagram).
Click [here](https://minuszerodegrees.net/ibm_xebec/ibm_xebec_cabling_info.htm) for a cabling information.
Click [here](https://minuszerodegrees.net/ibm_xebec/ibm_xebec_llf.htm) for information about low-level formatting.
Click [here](https://minuszerodegrees.net/ibm_xebec/ibm_xebec_pod.htm) for information about the controller's power-on diagnostics (POD).
Click [here](https://minuszerodegrees.net/rom/rom.htm) for the code contained in the two ROM's.

# [here](https://retrocomputing.stackexchange.com/questions/24383/ibm-xebec-xt-fixed-disk-adapter-custom-drive-types) 

it says that only 4 geometries were supported. Unable to determine which, seeking for information.

some info that is lacking [here](https://www.gleb.cz/ES1841/Xebec-3198-00XX-Info.htm) (in Russian, includes original disassembly of rom). long forum [thread](https://www.phantom.sannata.org/viewtopic.php?t=16892) in Russian.

**Comment by dbalsom** [comment]
The IBM/Xebec Variation #3 is what MartyPC emulates.  It only supports 10MB and 20MB drives, but I have hardcoded the emulated jumpers into the 20MB position so that's what you have to use for the moment.  MFM hard drives didn't get much bigger than that, at some point it makes more sense to implement an XTA/IDE controller like XTIDE

**Comment by darvin** [comment]
would pull requests with a hack way to increase the size of hard drive be accepted if's turnable by the config setting?

**Reply by dbalsom** [reply]
depends on how ugly of a hack it is. how were you planning it?
